### DISCUSSION TITLE
How to disable plugin autostart, and launch it on demand using 'RustAnalyzer start'

### DISCUSSION THREAD

**Discussion by mateuszaaa** [mrcjkb/rustaceanvim#226] [Q&A] [answered]
Hello, great work guys! Is it possible to not start lsp by default but rather manually with `RustAnalyzer start` ? I work on quite big rust projects and in most cases i dont want to run rust analyzer automatically

**Comment by mrcjkb** [comment, marked as answer]
Hey :wave:

There are two ways you can achieve this.

1. By setting the `server.auto_attach` option to `false` (see [advanced configuration](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#advanced-configuration)).
2. In case you use direnv or nix, this plugin won't start the LSP client if rust-analyzer isn't on the PATH.

To start rust-analyzer manually, use the `:RustAnalyzer start` command.

**Reply by UnkwUsr** [reply]
For those who come here after me: after disabling `auto_attach`, the command to start rustaceanvim manually is `:lua require('rustaceanvim.lsp').start()`

**Reply by mrcjkb** [reply]
`require('rustaceanvim.lsp').start()` is not part of the public API and may change.
There was a bug preventing the `:RustAnalyzer` command from being created if `auto_attach` is disabled. I've fixed that in the latest release (`8.0.0`). Please use the `:RustAnalyzer start` command to start rust-analyzer manually.
